client/my-sites/customer-home/components/celebrate-launch-modal.jsx

```diff
@@ -16,25 +18,44 @@ function CelebrateLaunchModal( { setModalIsOpen, site, allDomains } ) {
 	const translate = useTranslate();
 	const isPaidPlan = ! site?.plan?.is_free;
 	const isBilledMonthly = site?.plan?.product_slug?.includes( 'monthly' );
-
 	const transformedDomains = allDomains.map( createSiteDomainObject );
+	const [ isInitiallyLoaded, setIsInitiallyLoaded ] = useState( false );
 	const [ clipboardCopied, setClipboardCopied ] = useState( false );
 	const clipboardButtonEl = useRef( null );
 	const hasCustomDomain = Boolean(
 		transformedDomains.find( ( domain ) => ! domain.isWPCOMDomain )
 	);
+	const purchases = useSelector( ( state ) => getSitePurchases( state, site?.ID ) );
+	const actualPlanPurchase =
+		purchases
+			.filter(
+				( purchase ) => purchase.productSlug === site?.plan?.product_slug && purchase.active
+			)
+			.sort( ( a, b ) => new Date( a.expiryDate ) - new Date( b.expiryDate ) )[ 0 ] || null;
+	const isA4ASite = actualPlanPurchase?.partnerType === 'a4a_agency';
+	const isLoading = useSelector( isFetchingSitePurchases );
+
+	useEffect( () => {
+		dispatch( fetchSitePurchases( site?.ID ) );
+	}, [ dispatch, site?.ID ] );
+
+	useEffect( () => {
+		if ( ! isLoading && ( ! isPaidPlan || purchases.length > 0 ) && ! isInitiallyLoaded ) {
+			setIsInitiallyLoaded( true );
+		}
+	}, [ isLoading, purchases, isPaidPlan, isInitiallyLoaded ] );
```

Comment:
Looks like there are selectors (e.g. `getSitePlanSlug`) that we can leverage to simplify the code. 

```suggestion
	const sitePlanSlug = useSelector( ( state ) => getSitePlanSlug( state, site?.ID ) );
	const purchases = useSelector( ( state ) => getSitePurchases( state, site?.ID ) );
	const actualPlanPurchase = purchases.filter( ( purchase ) => purchase.productSlug === sitePlanSlug );
	const isA4ASite = isA4APurchase( actualPlanPurchase[0] );
```

Note that in the above suggestion, I've also introduced a helper function `isA4APurchase()` in `lib/purchases` to avoid sprinkling the string literal to various parts of the code base.

```
export function isA4APurchase( purchase?: Purchase ) {
	return 'a4a_agency' === purchase?.partnerType;
}
```

Reply:
I like the approach of introducing `isA4APurchase` that can be potentially reused later. It simplifies the whole logic as well. 👌🏼

Reply:
Pretty nice, indeed 👍